clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java

```diff
@@ -96,6 +96,14 @@ boolean filterByTTLandMaybeUpdateValue(final INPUT_VALUE value) {
     if (rmdTimestampType.equals(RmdTimestampType.VALUE_LEVEL_TIMESTAMP)) {
       return (long) rmdTimestampObject <= filterTimestamp;
     }
+    ByteBuffer valuePayload = getValuePayload(value);
+    if (valuePayload == null || valuePayload.remaining() == 0) {
+      /**
+       * We do not need check the field level TS for DELETE. If a field has TS older than TTL, it does not matter as it
+       * will still be overwritten by new PUT/UPDATE.
+       */
+      return false;
```

Comment:
I think we should still filter it out for Auto-purge use-cases. We should not even keep the keys around if a record is expected to be auto-purged

Reply:
Yeah, I was struggling on that to make a decision because I basically think that's fine. If you think we should be strict and clear about this I can update the PR.

Reply:
Isn't the behavior prior this is, the store can potentially drop the newer update depending on the delete timestamp in which case, the filtered record exists in the store as tombstone but now, the records will be filtered by the TTL push and the newer update from RT with older timestamp can potentially be accepted due to lack of RMD resulting in behavioral change?

Reply:
Hmm, my take is: 
1. Prior to this change, the push job will died because of exception. Not working at all. 
2. After the change, prior to Nisarg's suggestion: If final resolved value is a DELETE with higher TS than TTL, then it will be kept, and a PUT with TTL < TS < DELETE TS will still be dropped. If DELETE TS < TTL, for current code change it will be kept, but due to the fact that rewind PUT TS should be > TTL as how we configured TTL time and rewind time for TTL repush, then new PUT should be kept as expected.
3. If I take in Nisarg's change, it will try its best to remove DELETE with TS < TTL as well. This should look cleaner in theory.

Reply:
Addressed the issue in the latest commit. 
I thought about iterating over RMD timestamp fields but that looks not clean. 
Since it is offline flow logic, I ended up creating a default record with old DELETE timestamp for it to go through the below DELETE flow with TTL timestamp, which essentially the same. It is only used to determine whether such record should be kept or not, this default value won't be emitted as the result.